### PR TITLE
Fix orch return processing

### PR DESCRIPTION
**Pull request by roylarsen**
### What does this PR do?

Extends the Salt Runner processing to allow cloud tasks to be processed correctly.

### What issues does this PR fix or reference?

https://github.com/saltstack/salt/issues/50939

### Previous Behavior

Cloud functions were showing as succeeding.

### New Behavior

Runner processing will now report failures from cloud functions

### Tests written?

Yes

### Commits signed with GPG?

No

### CODE REVIEW

**Review by Ch3LL — CHANGES_REQUESTED** [review]
thanks for the pr! looks like there is a lint and test failure, if you could just clean those up. Thanks!

**Comment by roylarsen** [comment]
@Ch3LL - Sorry for the delay, I'll get those linting errors cleaned up and this re-submitted!

**Comment by pcn** [comment]
This seems like it's in-line with how e.g. the sls or highstate modules describe the run as successful only if all states have succeeded. It seems like it'd be a good idea to add docstrings to this effect in this case though, since it's new behaviour.

File: tests/unit/states/test_saltmod.py [hunk 1]
```diff
@@ -264,6 +264,28 @@ def test_runner(self):
...
+
+        ret = {'changes': {'return': {'servername': {'Error': 'servername already exists under mh-vmware:vmware'}}},
+               'name': 'cloud.profile', 'result': False,
+               'comment': 'Runner function \'cloud.profile\' failed.',
+               '__orchestration__': True}
+
+        runner_mock = MagicMock(return_value={'return': {'servername': {'Error': 'servername already exists under mh-vmware:vmware'}}})
+
+        with patch.dict(saltmod.__salt__, {'saltutil.runner': runner_mock}):
+          self.assertDictEqual(saltmod.runner(name), ret)
```
Comment (twangboy) [inline tests/unit/states/test_saltmod.py]: Indentation looks wrong here. Should be 4 spaces

File: tests/unit/states/test_saltmod.py [hunk 2]
```diff
@@ -264,6 +264,28 @@ def test_runner(self):
...
+
+        name = 'cloud.map_run'
+        ret = {'changes': {'return': {}}, 'name': 'cloud.map_run', 'result': False,
+               'comment': 'Runner function \'cloud.map_run\' failed.',
+               '__orchestration__': True}
+
+        runner_mock = MagicMock(return_value={'return': {}})
+
+        with patch.dict(saltmod.__salt__, {'saltutil.runner': runner_mock}):
+          self.assertDictEqual(saltmod.runner(name), ret)
```
Comment (twangboy) [inline tests/unit/states/test_saltmod.py]: Same here

**Review by twangboy — CHANGES_REQUESTED** [review]
This looks good. A few format errors. Also, we are no longer using the develop branch. All changes are being made on the master branch. Please rebase this against the master branch.

**Comment by sagetherage** [comment]
@roylarsen are you able to rebase this against master? Otherwise we may close it as abandoned.

**Comment by sagetherage** [comment]
@roylarsen are you able to rebase this against master?

**Comment by sagetherage** [comment]
Reviewed in Grooming 2021-MAY-03 closing

**Pull request closed without merging**